src/backends/backend.js

```diff
@@ -70,10 +71,11 @@ class Backend {
   currentUser() {
     if (this.user) { return this.user; }
     const stored = this.authStore && this.authStore.retrieve();
-    if (stored) {
+    if (stored && (stored.backend === this.backendName)) {
```

Comment:
No parens necessary on the equality check. Also, keep the naming synced - if it's `this.backendName`, let's call it `backendName` in the stored user.